nvflare/lighter/impl/workspace.py

```diff
@@ -43,50 +44,56 @@ def __init__(self, template_file):
               wip/  <--- this is only used during runtime, and will be removed when the provision command exits
 
         Args:
-            template_file: name(s) of template file(s) containing scripts and configs to put into startup folders
+            template_file: one or more template file names
         """
         self.template_files = template_file
 
-    def _make_dir(self, dirs):
-        for dir in dirs:
-            if not os.path.exists(dir):
-                os.makedirs(dir)
+    def _build_template(self, ctx: ProvisionContext):
+        prov_folder = os.path.dirname(prov.__file__)
+        temp_folder = os.path.join(prov_folder, "templates")
 
-    def initialize(self, ctx):
-        workspace_dir = ctx["workspace"]
+        temp_files_to_load = self.template_files
+        if not temp_files_to_load:
+            # load everything
+            temp_files_to_load = [f for f in os.listdir(temp_folder) if os.path.isfile(f)]
+        elif isinstance(temp_files_to_load, str):
+            temp_files_to_load = [temp_files_to_load]
+
+        template = dict()
+        for f in temp_files_to_load:
+            template.update(load_yaml(os.path.join(temp_folder, f)))
```

Comment:
depends on the order of "temp_files_to_load", the latter file might overwrite the previous file if they have same keys

Reply:
That's why I at least move all section keys into the constants so dups can be detected.
Currently, the 3 temp files don't have conflicts. But in future, we should introduce naming convention. For example, all keys in a file should start with a unique prefix.